### PR TITLE
Add tests and refactor

### PR DESCRIPTION
**Pull request by jonathanlloyd**

### CODE REVIEW

File: src/PushNotifications.php [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
+/**
+ * Pusher Push Notifications client class
+ * Used to publish notifications to the Pusher Push Notifications API
+ * http://www.pusher.com/push-notifications
+ */
 class PushNotifications {
   const SDK_VERSION = "0.10.5";
+  const MAX_INTERESTS = 100;
+  const MAX_INTEREST_LENGTH = 164;
+  const INTEREST_REGEX = "/^(_|-|=|@|,|\\.|:|[A-Z]|[a-z]|[0-9])*$/";
```
Comment (luismfonseca) [inline src/PushNotifications.php]: I think it should end with `+$`?

File: src/PushNotifications.php [hunk 2]
```diff
@@ -1,12 +1,31 @@
...
+
+  private $options = array();
+  private $client = null;
 
-  public function __construct($options) {
+  public function __construct($options, $client = null) {
```
Comment (luismfonseca) [inline src/PushNotifications.php]: Hmm, couldn't the client be part of the `options`?

TBH, I'd like more if the `instanceId` and `secretKey` were dedicated arguments and then just a misc `options` for the rest.

File: composer.json [hunk 3]
```diff
@@ -2,7 +2,15 @@
     "name": "pusher/pusher-push-notifications",
     "version": "0.10.5",
     "license": "MIT",
-    "require": {},
+    "require": {
+        "php": ">=5.6.0",
+        "guzzlehttp/guzzle": "~6.0"
```
Comment (luismfonseca) [inline composer.json]: We are adding an extra dependency here, do we really benefit using this over plain curl?

File: src/PushNotifications.php (hunk 2 above)
Comment (jonathanlloyd) [reply src/PushNotifications.php]: I'd like to do this in a separate PR as it would be a breaking change.

File: composer.json (hunk 3 above)
Comment (jonathanlloyd) [reply composer.json]: We are actually exchanging one dependency for another - we no longer require the PHP cURL extension!

**Pull request merged**